block/manager.go

```diff
@@ -755,9 +746,30 @@ func (m *Manager) getCommit(header types.Header) (*types.Commit, error) {
 	// note: for compatibility with tendermint light client
 	consensusVote := header.MakeCometBFTVote()
 
-	sign, err := m.proposerKey.Sign(consensusVote)
-	if err != nil {
-		return nil, err
+	var (
+		sign []byte
+		err  error
+	)
+	switch m.proposerKey.(type) {
+	case *crypto.Ed25519PrivateKey:
+		sign, err = m.proposerKey.Sign(consensusVote)
+		if err != nil {
+			return nil, err
+		}
+	case *crypto.Secp256k1PrivateKey:
```

Comment:
As the sign implementation differs from `crypto.Secp256k1PrivateKey.Sign()`, please comment why this is needed.
Doesn't this break compatibility for users wishing to use secp256k1 from cosmos-sdk?

Reply:
mainly because of the difference in `Sign` functions between cometbft/secp256k1 and golibp2p2/secp256k1. the cometbft clients (omni evm) calls `VerifySignature` of cometbft and signing using `golibp2p2/secp256k1` will not pass, hence I had to mimic the cometbft/secp256k1 sign method. see: https://github.com/cometbft/cometbft/blob/main/crypto/secp256k1/secp256k1.go#L128 and https://github.com/libp2p/go-libp2p/blob/master/core/crypto/secp256k1.go#L75 for further details. Note that, sign function is same for ed25519 hence it was working fine before.